### DISCUSSION TITLE
Formula dimension change in application molecule builder

### DISCUSSION THREAD

**Discussion by moPellowe** [Open-Systems-Pharmacology/Forum#1275] [Questions & Problems] [answered]
When I create a formula in the application molecule builder (as in the image below), is there a way to change the dimension of the formula? I'm getting an error because the formula I'm using evalutes to an amount but the formula dimension is "Dimensionless" by default and I'm not sure how to change it.

![image](https://user-images.githubusercontent.com/57540473/183969010-803e4715-a511-4386-87d0-1f4e1aec7327.png)

**Comment by PavelBal** [comment]
Unfortunately, you cannot change the dimension (I made a feature request for this a while ago..). Workaround - assign the formula to a parameter, then change the dimension of the parameter - this will also change the dimension of the formula.

**Reply by moPellowe** [reply]
Thank you for your response. I tried your workaround, but it didn't work for me. Do you mean to change the parameter within the Formulas tab or the actual parameter wherever it is contained?

**Comment by PavelBal** [comment, marked as answer]
Created a new assigment with the formula:

![image](https://user-images.githubusercontent.com/25903040/184117204-4ef36639-ce59-4052-bfad-33c8ad20bdfe.png)

As in your case, the dimension is "Dimensionless":

![image](https://user-images.githubusercontent.com/25903040/184117267-e822b9e6-053e-4b30-b043-6981bb0ac75f.png)


Now the workaround. Create any parameter that is "Dimensionless" and assign the formula to it:

![image](https://user-images.githubusercontent.com/25903040/184117440-2a88c1df-5df8-4b13-9674-e161413328d3.png)


Now change the dimension of this parameter:

![image](https://user-images.githubusercontent.com/25903040/184117498-77a07419-d871-489b-9250-bf3f14312481.png)

The dimension of the formula also changed: (you might need to close and open the events building block to see the update, it is a GUI refresh problem)

![image](https://user-images.githubusercontent.com/25903040/184117634-9340ef11-7500-4b78-8b67-661fc935f225.png)

**Reply by moPellowe** [reply]
Thank you so much for the step-by-step instructions. I believe this should work.
